src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -109,14 +135,28 @@ private ValueTask<FlushResult> WriteCore(HubMessage message)
         {
             try
             {
-                // This will internally cache the buffer for each unique HubProtocol
-                // So that we don't serialize the HubMessage for every single connection
-                var buffer = message.WriteMessage(Protocol);
+                // We know that we are only writing this message to one receiver, so we can
+                // write it without caching.
+                Protocol.WriteMessage(message, _connectionContext.Transport.Output);
```

Comment:
Love it

Reply:
Users have to know to use `SerializedHubMessage` for all `HubMessage` for one.

Reply:
To be fair, it's not that it doesn't work if they use the wrong thing, it's just that they only get the optimized serialization if they use `SerializedHubMessage`